src/openapi/streaming/streaming.js

```diff
@@ -523,6 +525,103 @@ Streaming.prototype.disposeSubscription = function(subscription) {
     }
 };
 
+function handleSubscriptionReadyForUnsubscribe(subscriptions, resolve) {
+    let allSubscriptionsReady = true;
+    for (let i = 0; i < subscriptions.length && allSubscriptionsReady; i++) {
+        if (!subscriptions[i].isReadyForUnsubscribeByTag()) {
+            allSubscriptionsReady = false;
+        }
+    }
+
+    if (allSubscriptionsReady) {
+        resolve();
+    }
+}
+
+function getSubscriptionsToRemove(serviceGroup, url, tag) {

```

Comment:
Better name for this function would be getSubscriptionsByTag. It doesn't have any filtering logic that would indicate that  given subscription is pending for removal,  and that's what i though when i first read this initial name.